lib/generator.js

```diff
@@ -82,10 +91,12 @@ class Generator {
    * @param {Boolean} [options.debug=false] Enable more specific errors in the console. At the moment it only shows specific errors about filters. Keep in mind that as a result errors about template are less descriptive.
    */
   constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false } = {}) {
+    const unexpectedOptionsError = validateOptions(arguments[arguments.length - 1] || []);
+    if (arguments.length === 3 && unexpectedOptionsError) throw new Error(unexpectedOptionsError);
```

Comment:
damn I lost one comment, let we write again :( 

what do you think about splitting responsibility of `validateOptions` so it is only responsible to `getInvalidOptions` and the building of the error string stays in generator? then you can avoid things like `arguments.length === 3`

something like:
```
const invalidOptions = getInvalidOptions(arguments[arguments.length - 1] || [])
if (invalidOptions.length) throw new Error(`These options are not supported by the generator: ${invalidOptions.join(', ')}`);
```

I'm super curious what do you think about it

Reply:
Nice one, I thought the same.

As per the `arguments.length === 3`, it was there to avoid throwing a new error in case a third argument isn't given, as `Object.keys()` is receiving the last argument always and when a string is given it returns an array of strings corresponding to its length.

I'm gonna figure out how to return just the invalid options, which would make it easier 👌.